application/frontend/tests/auto-mine.test.ts

```diff
@@ -0,0 +1,86 @@
+import { test, Page } from '@playwright/test';
+
+// Helper functions from manual crafting remain the same
+async function getGameCanvas(page: Page) {
+    const canvas = await page.waitForSelector('#game-container canvas', { timeout: 10000 });
+    await page.waitForTimeout(2000);
+    return canvas;
+}
+
+function calculateScreenPosition(
+    position: { x: number; y: number },
+    bounds: { x: number; y: number },
+    scale: { x: number; y: number },
+    zoomLevel: number
+) {
+    return {
+        x: bounds.x + (position.x * scale.x * zoomLevel),
+        y: bounds.y + (position.y * scale.y * zoomLevel)
+    };
+}
+
+// Test automated mining using MinerPlacementScene
+test.describe('Automated Mining Test', () => {
+    test('should complete automated mining workflow', async ({ page }) => {
+        await page.goto('http://localhost:5173/game');
+        const canvas = await getGameCanvas(page);
+        const bounds = await canvas.boundingBox();
+
+        if (!bounds) {
+            throw new Error('Could not get canvas bounds');
+        }
+
+        const displaySize = await page.evaluate(() => {
+            const canvas = document.querySelector('#game-container canvas');
+            if (!canvas) return null;
+            return {
+                width: canvas.clientWidth,
+                height: canvas.clientHeight,
+            };
+        });
+
+        console.log('Canvas display size:', displaySize);
+        if (!displaySize) {
+            throw new Error('Could not get canvas display size');
+        }
+
+        const scale = {
+            x: displaySize.width / 2048,
+            y: displaySize.height / 1152,
+          };
+
+
+        // 1: Open Build Menu
```

Comment:
Can you separate each of these actions into an individual `test()` call to annotate them?